src/librustdoc/html/static/js/main.js

```diff
@@ -329,6 +329,26 @@ function preLoadCss(cssUrl) {
             search.innerHTML = "<h3 class=\"search-loading\">" + searchState.loadingText + "</h3>";
             searchState.showResults(search);
         },
+        descShards: new Map(),
+        loadDesc: async function({descShard, descIndex}) {
+            if (descShard.promise === null) {
+                descShard.promise = new Promise((resolve, reject) => {
```

Comment:
I don't understand how the promise is working here. There is nothing to await, or did I miss something? We could update `loadScript` to make it `async` too.

Reply:
Thanks for adding the comment. :+1: